File: api/pkg/responses/system.go
package responses

type SystemInfo struct {
	Version        string                    `json:"version"`
	Endpoints      *SystemEndpointsInfo      `json:"endpoints"`
	Setup          bool                      `json:"setup"`
	Authentication *SystemAuthenticationInfo `json:"authentication"`
}

type SystemAuthenticationInfo struct {
	Local bool `json:"local"`
	SAML  bool `json:"saml"`
}

type SystemEndpointsInfo struct {
	API string `json:"api"`
	SSH string `json:"ssh"`
}

File: ui-react/apps/console/src/types/system.ts
export interface SystemInfo {
  version: string;
  setup: boolean;
  authentication: {
    local: boolean;
    saml: boolean;
  };
  endpoints: {
    ssh: string;
    api: string;
  };
}

export interface SetupRequest {
  name: string;
  username: string;
  email: string;
  password: string;
}
